### PR TITLE
Fixing JIRA DS-1189-Added Legacy timezone support

### PR DESCRIPTION
**Pull request by anupama-pathirage**

### CODE REVIEW

File: components/data-services/org.wso2.carbon.dataservices.common/src/main/java/org/wso2/carbon/dataservices/common/DBConstants.java [hunk 1]
```diff
@@ -391,6 +391,9 @@ private RDBMS() {
         public static final String DYNAMIC_USER_AUTH_CLASS = "dynamicUserAuthClass";
         public static final String DYNAMIC_USER_AUTH_MAPPING = "dynamicUserAuthMapping";
         public static final String USERNAME_WILDCARD = "*";
+        public static final String DSS_TIMERZONE= "dss.timezone";
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.common/src/main/java/org/wso2/carbon/dataservices/common/DBConstants.java]: Need to add a space before =

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 2]
```diff
@@ -197,8 +199,22 @@ public void init(String query) throws DataServiceFault {
...
          * to use when setting timestamp for prepared statements
          * and retrieving timestamps from result sets
+         * This time conversion feature can be enabled/disabled
+         * using -Ddss.legacy.timezone.mode=true or -Ddss.legacy.timezone.mode=false.
+         * Default values are dss.legacy.timezone.mode=false with default timezone "UTC"
+         * This default time zone can be set via -Ddss.timezone=
          */
-        calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        String userTimeZone = System.getProperty(RDBMS.DSS_TIMERZONE);
+        if(userTimeZone == null || userTimeZone.isEmpty()) {
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Need a space after if

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 3]
```diff
@@ -197,8 +199,22 @@ public void init(String query) throws DataServiceFault {
...
+            userTimeZone = RDBMS.TIMEZONE_UTC;
+        }
+        calendar = Calendar.getInstance(TimeZone.getTimeZone(userTimeZone));
+
+        String legacyTimezoneMode = System.getProperty(RDBMS.DSS_LEGACY_TIMEZONE_MODE);
+        if("true".equalsIgnoreCase(legacyTimezoneMode)) {
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Need a space after if

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 4]
```diff
@@ -1175,7 +1191,11 @@ private DataEntry getDataEntryFromRS(ResultSet rs) throws SQLException {
                 paramValue = new ParamValue(value);
                 break;
             case Types.TIMESTAMP:
-                sqlTimestamp = rs.getTimestamp(i, calendar);
+                if(timeConvertEnabled) {
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Need a space after if

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 5]
```diff
@@ -1688,20 +1708,32 @@ private void setTimestampValue(int queryType, String paramName, String value, St
...
+                    } else {
+                        sqlQuery.setTimestamp(i + 1, timestamp);
+                    }
                 }
             } else {
                 if (value == null) {
                     ((CallableStatement) sqlQuery).setNull(i + 1, java.sql.Types.TIMESTAMP);
                 } else {
-                    ((CallableStatement) sqlQuery).setTimestamp(i + 1, timestamp, calendar);
+                    if(timeConvertEnabled) {
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Need a space after if

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 6]
```diff
@@ -1688,20 +1708,32 @@ private void setTimestampValue(int queryType, String paramName, String value, St
...
+                        ((CallableStatement) sqlQuery).setTimestamp(i + 1, timestamp);
+                    }
                 }
             }
         } else if ("INOUT".equals(paramType)) {
             if (value == null) {
                 ((CallableStatement) sqlQuery).setNull(i + 1, java.sql.Types.TIMESTAMP);
             } else {
-                ((CallableStatement) sqlQuery).setTimestamp(i + 1, timestamp, calendar);
+                if(timeConvertEnabled) {
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Need a space after if

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 7]
```diff
@@ -2058,7 +2090,11 @@ private ParamValue getOutparameterValue(CallableStatement cs, String type, int o
                 return new ParamValue(elementValue == null ? null
                         : ConverterUtil.convertToString((Date) elementValue));
             } else if (type.equals(DBConstants.DataTypes.TIMESTAMP)) {
-                elementValue = cs.getTimestamp(ordinal, calendar);
+                if(timeConvertEnabled) {
```
Comment (Manuri) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Need a space after if

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 2 above)
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Please fix code formatting, fix in all the places.

File: components/data-services/org.wso2.carbon.dataservices.common/src/main/java/org/wso2/carbon/dataservices/common/DBConstants.java (hunk 1 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.common/src/main/java/org/wso2/carbon/dataservices/common/DBConstants.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 2 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 3 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 4 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 5 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 6 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 7 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 2 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java [hunk 8]
```diff
@@ -1688,20 +1708,32 @@ private void setTimestampValue(int queryType, String paramName, String value, St
                 if (value == null) {
                     sqlQuery.setNull(i + 1, java.sql.Types.TIMESTAMP);
                 } else {
-                    sqlQuery.setTimestamp(i + 1, timestamp, calendar);
+                    if(timeConvertEnabled) {
```
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: Fixed with PR: #129

**Comment by mahaboob** [comment]
Can someone please help me how to handle timezone issue with DSS 
https://stackoverflow.com/questions/46596508/wso2-dss-timezone-issue

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java (hunk 2 above)
Comment (mahaboob) [reply components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/SQLQuery.java]: https://stackoverflow.com/questions/46596508/wso2-dss-timezone-issue
can you please let me know how to fix this issue

**Pull request merged**